File: AddressBook/AddressBook/Dtos/PersonEmail.cs
using AddressBook.Entities;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using AddressBook.Common.Paging;

namespace AddressBook.Dtos
{
    public class PersonEmailRes
    {
        public int PersonEmailId { get; set; }
        public string? Email { get; set; }

        public int? PersonId { get; set; }
        public PersonRes? Person { get; set; }
        public int? EmailLabelId { get; set; }
        public EmailLabelRes? EmailLabel { get; set; }
    }

    public class PersonEmailReqEdit
    {
        [Required]
        public string? Email { get; set; }

        public int? PersonId { get; set; }
        public int? EmailLabelId { get; set; }
    }

    public class PersonEmailReqSearch : PagedReq
    {
        public int? PersonId { get; set; }
    }
}

File: AddressBook/react-client/src/dtos/PersonEmail.ts
import Common from "../utility/Common";
import { EmailLabelRes } from "./EmailLabel";
import { PagedReq } from "./PagedReq";
import { PersonRes } from "./Person";

export interface PersonEmailRes {
  personEmailId?: number;
  email?: string;
  personId?: number;
  person?: PersonRes;
  emailLabelId?: number;
  emailLabel?: EmailLabelRes;
}

export class PersonEmailReqEdit {
  email?: string = "";
  personId?: number = 0;
  emailLabelId?: number = 0;
}

export class PersonEmailReqSearch extends PagedReq {
  personId?: string;
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy,
      searchText = "",
    }: PagedReq,
    {personId = ""}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
    this.personId = personId;
  }
}